helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -140,6 +160,81 @@ public static String serializeByComma(List<String> objects) {
     }
   }
 
+  /**
+   * Returns the expected ideal ResourceAssignments for the given resources in the cluster
+   * calculated using the read-only WAGED rebalancer.
+   * @param metadataStoreAddress
+   * @param clusterConfig
+   * @param instanceConfigs
+   * @param liveInstances
+   * @param newIdealStates
+   * @param newResourceConfigs
+   * @return
+   */
+  public static Map<String, ResourceAssignment> getIdealAssignmentForWagedFullAuto(
```

Comment:
I feel RebalanceUtil is a better placement, although the existing method is already in the HelixUtil. What do you think?

Reply:
If it will be exposed to customers, let's put it HelixUtil. It is better to have some centralized place for user using that.

Reply:
Already reviewed that option, but if we were to place this in RebalanceUtil, we should move the existing function there too, but that would cause a backward-incompatible change. Let's keep it in here for consistency.